cdp/smart_contract.py

```diff
@@ -373,6 +414,105 @@ def read(
         )
         return cls._convert_solidity_value(model)
 
+    @classmethod
+    def update(
```

Comment:
Update should be an instance method.

E.g. we've already registered or fetched a SmartContract resource that we then can call update on